src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs

```diff
@@ -225,12 +225,19 @@ internal void ToString(TraceFormat traceFormat, StringBuilder sb)
                     Type? declaringType = mb.DeclaringType;
                     string methodName = mb.Name;
                     bool methodChanged = false;
-                    if (declaringType != null && declaringType.IsDefined(typeof(CompilerGeneratedAttribute), inherit: false))
+
+                    // Removing Debugger support with the ILLinker will remove CompilerGeneratedAttribute.
+                    // But accessing CompilerGeneratedAttribute in code will force the linker to keep it.
+                    // So only access CompilerGeneratedAttribute when Debugger.IsSupported is true.
+                    if (Debugger.IsSupported)
                     {
-                        isAsync = typeof(IAsyncStateMachine).IsAssignableFrom(declaringType);
```

Comment:
Won't this change the stacktrace format ?

Reply:
Correct, it does. See https://github.com/dotnet/runtime/pull/39000#discussion_r452362679 for discussion.